src/proxy/IMap.ts

```diff
@@ -341,7 +341,7 @@ export interface IMap<K, V> extends DistributedObject {
      * Locks the given key for this map. Promise is resolved when lock is
      * successfully acquired. This means it may never be resolved if some
      * other process holds the lock and does not unlock it. A lock may be
-     * acquired on non-existent keys. Other processes wait on non-existent key.
+     * acquired on non-existent keys. Other processes wait on the locked non-existent key.
```

Comment:
Do not have strong opinion here. I did not understand this non-existent key thing. Could anyone explain?

Reply:
What about writing the same thing here as well? Certainly this comment is less understandable due to its client centric perspective

Reply:
It makes sense, the writer of this jsdoc probably didn't copy directly since the Java side documentation is mentioning threads. What do you think about replacing this part with something like
```
A lock may be acquired on non-existent keys. In that case, other clients would block until the non-existent key is unlocked. If the lock holder introduces the key to the map, the put operation is not blocked. If a client not holding a lock on the non-existent key tries to introduce the key while a lock exists on the non-existent key, the put operation blocks until it is unlocked.
```

Reply:
Seems ok to me a question:
>  If the lock holder introduces the key to the map, the put operation is not blocked
Did not get if this "block" is for lock holder, or other clients?

Reply:
The lock holder will not block but others will